File: parse-logs.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('Logs2.md', 'utf8'));

// Regex to strip ANSI escape codes
const stripAnsi = (str) => str.replace(/\x1b\[[0-9;]*m/g, '');

const out = data.map(log => {
    const time = new Date(log.timestamp).toISOString();
    const severity = log.severity || 'INFO';
    const text = stripAnsi(log.textPayload || '');
    return `[${time}] [${severity}] ${text}`;
}).join('\n');

fs.writeFileSync('Logs2_parsed.txt', out);
console.log('Parsed successfully with ANSI stripped, lines:', data.length);

File: parse_logs.py
import json
import os

input_file = r'c:\Users\Sherry\Documents\Convergent_AI\Logs2.md'
output_file = r'c:\Users\Sherry\Documents\Convergent_AI\Logs2_parsed.txt'

try:
    with open(input_file, 'r', encoding='utf-8') as f:
        data = json.load(f)
        
    with open(output_file, 'w', encoding='utf-8') as out:
        for entry in data:
            timestamp = entry.get('timestamp', '')
            text = entry.get('textPayload', '')
            if timestamp and text:
                out.write(f"[{timestamp}] {text}\n")
            elif text:
                out.write(f"{text}\n")
    print(f"Successfully parsed {len(data)} logs to {output_file}")
except Exception as e:
    print(f"Error: {e}")
